sklearn/isotonic.py

```diff
@@ -259,7 +259,11 @@ def _build_y(self, X, y, sample_weight, trim_duplicates=True):
             sample_weight = np.ones(len(y), dtype=X.dtype)
 
         order = np.lexsort((y, X))
-        X, y, sample_weight = [array[order] for array in [X, y, sample_weight]]
+
+        # set everything to the same dtype to avoid errors in _make_unique()
+        X_dtype = X.dtype
+        X, y, sample_weight = [array[order].astype(X_dtype, copy=False)
```

Comment:
```suggestion
        X, y, sample_weight = [array[order].astype(X.dtype, copy=False)
```

and remove the `X_dtype` variable.

Reply:
see https://github.com/scikit-learn/scikit-learn/issues/13503 for explanation of problem